pkg/nodeagent/dbus/dbus.go

```diff
@@ -0,0 +1,158 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dbus
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/coreos/go-systemd/v22/dbus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/record"
+)
+
+const done = "done"
+
+type Dbus interface {
+	// Enable the given units, same as executing systemctl enable unit
+	Enable(ctx context.Context, unitNames ...string) error
+	// Disable the given units, same as executing systemctl disable unit
+	Disable(ctx context.Context, unitNames ...string) error
+	// Start the given unit and record an event to the node object, same as executing systemctl start unit
+	Start(ctx context.Context, recorder record.EventRecorder, node *corev1.Node, unitName string) error
+	// Stop the given unit and record an event to the node object, same as executing systemctl stop unit
+	Stop(ctx context.Context, recorder record.EventRecorder, node *corev1.Node, unitName string) error
+	// Restart the given unit and record an event to the node object, same as executing systemctl restart unit
+	Restart(ctx context.Context, recorder record.EventRecorder, node *corev1.Node, unitName string) error
+	// DaemonReload reload the systemd configuration, same as executing systemctl daemon-reload
+	DaemonReload(ctx context.Context) error
+}
+type db struct{}
+
+func New() Dbus {
+	return &db{}
+}
+
+func (*db) Enable(ctx context.Context, unitNames ...string) error {
+	dbc, err := dbus.NewWithContext(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to connect to dbus: %w", err)
+	}
+	defer dbc.Close()
+
+	_, _, err = dbc.EnableUnitFilesContext(ctx, unitNames, false, true)
+	return err
+}
+
+func (*db) Disable(ctx context.Context, unitNames ...string) error {
+	dbc, err := dbus.NewWithContext(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to connect to dbus: %w", err)
+	}
+	defer dbc.Close()
+
+	_, err = dbc.DisableUnitFilesContext(ctx, unitNames, false)
+	return err
+}
+
+func (*db) Stop(ctx context.Context, recorder record.EventRecorder, node *corev1.Node, unitName string) error {
+	dbc, err := dbus.NewWithContext(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to connect to dbus: %w", err)
+	}
+	defer dbc.Close()
+
+	c := make(chan string)
+
+	if _, err := dbc.StopUnitContext(ctx, unitName, "replace", c); err != nil {
+		return fmt.Errorf("unable to stop unit %s: %w", unitName, err)
+	}
+
+	job := <-c
+	if job != done {
```

Comment:
Can you please inline the assignment into the `if` statement?

```suggestion
	if job := <-c; job != done {
```